activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -564,18 +598,24 @@ def default_insert_value(column) # :nodoc:
         private_constant :DEFAULT_INSERT_VALUE
 
         # Lowest level way to execute a query. Doesn't check for illegal writes, doesn't annotate queries, yields a native result object.
-        def raw_execute(sql, name = nil, binds = [], prepare: false, async: false, allow_retry: false, materialize_transactions: true, batch: false)
-          type_casted_binds = type_casted_binds(binds)
-          log(sql, name, binds, type_casted_binds, async: async, allow_retry: allow_retry) do |notification_payload|
-            with_raw_connection(allow_retry: allow_retry, materialize_transactions: materialize_transactions) do |conn|
-              result = perform_query(conn, sql, binds, type_casted_binds, prepare: prepare, notification_payload: notification_payload, batch: batch)
-              handle_warnings(result, sql)
+        def raw_execute(intent)
+          # Handle Arel compilation if needed
+          compile_arel_in_intent(intent)
```

Comment:
I may be wrong, but is this an unintended API change? I think that `#raw_execute` did not previously accept Arel, but only raw sql? And it looks like you changed all the internal code to compile Arel into sql before calling `#raw_execute`.